conformance/tests/httproute-redirect-port-and-scheme.go

```diff
@@ -0,0 +1,298 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	"sigs.k8s.io/gateway-api/conformance/utils/http"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+	"sigs.k8s.io/gateway-api/conformance/utils/suite"
+	"sigs.k8s.io/gateway-api/conformance/utils/tls"
+)
+
+func init() {
+	ConformanceTests = append(ConformanceTests, HTTPRouteRedirectPortAndScheme)
+}
+
+var HTTPRouteRedirectPortAndScheme = suite.ConformanceTest{
+	ShortName:   "HTTPRouteRedirectPortAndScheme",
+	Description: "An HTTPRoute with port and scheme redirect filter",
+	Manifests:   []string{"tests/httproute-redirect-port-and-scheme.yaml"},
+	Features: []suite.SupportedFeature{
+		suite.SupportGateway,
+		suite.SupportHTTPRoute,
+		suite.SupportHTTPRoutePortRedirect,
+	},
+	Test: func(t *testing.T, suite *suite.ConformanceTestSuite) {
+		ns := "gateway-conformance-infra"
+
+		gwNN := types.NamespacedName{Name: "same-namespace", Namespace: ns}
+		routeNN := types.NamespacedName{Name: "http-route-for-listener-on-port-80", Namespace: ns}
+		gwAddr80 := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		gwNN = types.NamespacedName{Name: "same-namespace-with-http-listener-on-8080", Namespace: ns}
+		routeNN = types.NamespacedName{Name: "http-route-for-listener-on-port-8080", Namespace: ns}
+		gwAddr8080 := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		gwNN = types.NamespacedName{Name: "same-namespace-with-https-listener", Namespace: ns}
+		routeNN = types.NamespacedName{Name: "http-route-for-listener-on-port-443", Namespace: ns}
+		gwAddr443 := kubernetes.GatewayAndHTTPRoutesMustBeAccepted(t, suite.Client, suite.TimeoutConfig, suite.ControllerName, kubernetes.NewGatewayRef(gwNN), routeNN)
+		kubernetes.HTTPRouteMustHaveResolvedRefsConditionsTrue(t, suite.Client, suite.TimeoutConfig, routeNN, gwNN)
+
+		certNN := types.NamespacedName{Name: "tls-validity-checks-certificate", Namespace: string(ns)}
+		cPem, keyPem, err := GetTLSSecret(suite.Client, certNN)
+		if err != nil {
+			t.Fatalf("unexpected error finding TLS secret: %v", err)
+		}
+
+		////////////////////////////////////////////////////////////////////////////
+		// Test cases that use http-route-for-listener-on-port-80
+		////////////////////////////////////////////////////////////////////////////
+
+		testCases := []http.ExpectedResponse{
```

Comment:
Nit: Somewhere in here, maybe above this section, it would be helpful to add a note clearly stating that wherever port is not specified by a test case, it is still valid and acceptable for a port to be specified in the redirect if it corresponds to the scheme (80 and 443). (Essentially the work you did in https://github.com/kubernetes-sigs/gateway-api/pull/1880/commits/fa0691a28576bdad7a163dff31d7d278747bbced).